internal/datastore/common/sql.go

```diff
@@ -683,6 +737,40 @@ func (exc QueryRelationshipsExecutor) ExecuteQuery(
 
 	// Add FROM clause.
 	from := query.schema.RelationshipTableName
+	if query.fromTable != "" {
+		from = query.fromTable
+	}
+
+	// Add index hints, if any.
+	if query.indexingHint != nil {
+		// Check for a SQL prefix (pg_hint_plan).
+		sqlPrefix, err := query.indexingHint.SQLPrefix()
```

Comment:
add validation that only one of these is set, since they are datastore specific.

Reply:
They are right now, but that may not be the case in the future. It should be up to the caller to decide IMO and the caller's tests would validate